superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx

```diff
@@ -1252,6 +1430,24 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
                 />
                 {renderPreferredSelector()}
                 {renderAvailableSelector()}
+                <Upload
+                  name="databaseFile"
+                  id="databaseFile"
+                  data-test="database-file-input"
+                  accept=".yaml,.json,.yml,.zip"
+                  // upload is handled by hook
+                  customRequest={() => {}}
+                  onChange={info => onDbImport(info)}
```

Comment:
onChange={onDbImport}

Reply:
`onDbImport` has to have `info` passed in as a parameter to work

Reply:
Nevermind actually! I tried it locally just to see and I _don't_ have to pass the param like I thought. I learned something today haha! Does that work for any function passed in like this? Or does it only work if it takes just one parameter?

Fixed in [`this commit`](https://github.com/apache/superset/pull/19314/commits/71ebaf325d3c185a083a1f3c0a2818deb5d75208)

Reply:
@lyndsiWilliams @hugh's suggestion does that! :)

The best way to think of this IMHO is, when you write:

```js
onChange={info => onDbImport(info)}
```

You're assigning an anonymous function (`info => onDbImport(info)`) to `onChange`. What does that anonymous function do? It takes an `info` and returns the result of calling `onDbImport(info)`. But... isn't that the exact same thing that `onDbImport` does? Regardless of the value of `info`, passing it to the anonymous function returns the same result of pasing it to `onDbImport`, so they're identical.

which means you can replace `info => onDbImport(info)` with `onDbImport`:

```js
onChange={onDbImport}
```

Reply:
Ohhh okay that makes sense! Thank you for the explanation! 😁